tests/functional/testSubscription.js

```diff
@@ -1486,10 +1483,125 @@ describe('Subscription view', function() {
              .body = SUBSCRIPTION_DETAILS;
        };
 
-       await clickSettingsIcon();
        await openSubscriptionManagement();
 
        await vpn.waitForQuery(
            queries.screenSettings.subscriptionView.ANNUAL_UPGRADE.hidden());
      });
+
+  describe('Subscription management telemetry tests', () => {
+
+    const testCases = [["annual plan", "account"], ["monthly plan", "account_with_change_plan"]];
+
+    testCases.forEach(([testCase, subscriptionManagementScreenTelemetryId]) => {
+      
+      describe(`Subscription management view events are recorded (${testCase})`, () => {
+
+        const isMonthlyPlanTest = testCase === "monthly plan"
+
+        //Check if we are testing the monthly plan so we can mock the guardian response
+        beforeEach(async () => {
```

Comment:
@brizental do you know of any good way to do this before each item in `testCases` instead of before each test (it)?